aten/src/ATen/native/TensorIterator.h

```diff
@@ -333,6 +338,7 @@ struct CAFFE2_API TensorIterator {
   bool resize_outputs_ = true;
   bool is_reduction_ = false;
   bool compute_common_dtype_ = true;
+  bool compute_common_dtype_only_for_inputs_ = false;
```

Comment:
Having both `compute_common_dtype_` and `compute_common_dtype_only_for_inputs_` is a little redundant and makes the code harder to read. When `compute_common_dtype` is false, does `compute_common_dtype_only_for_inputs_` do anything by itself?

I think what we're trying to express here is something like an enum: compute_common_dtype can either be `COMPUTE_ALL` (to compute over inputs & outputs), `COMPUTE_INPUTS` (to compute over inputs) or `COMPUTE_NONE` (don't compute any common dtype).

Reply:
I think it's a good idea.